src/ASM/LR/ASMu2Dmx.C

```diff
@@ -182,6 +182,17 @@ bool ASMu2Dmx::generateFEMTopology ()
     m_basis.resize(vec.size());
     for (size_t i=0;i<vec.size();++i)
       m_basis[i].reset(new LR::LRSplineSurface(vec[i].get()));
+
+    // we need to project on something that is not one of our bases
+    if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
+        ASMmxBase::Type == ASMmxBase::DIV_COMPATIBLE ||
+        ASMmxBase::Type == ASMmxBase::SUBGRID) {
+      auto vec2 = ASMmxBase::establishBases(tensorspline,
+                                            ASMmxBase::FULL_CONT_RAISE_BASIS1);
+      projBasis.reset(new LR::LRSplineSurface(vec2.front().get()));
```

Comment:
And here, since we only need the first component:
```
Go::SplineSurface* srf = ASMmxBase::establishBases(tensorspline,
                                                   ASMmxBase::FULL_CONT_RAISE_BASIS1).front().get();
projBasis.reset(new LR::LRSplineSurface(srf));
```
or maybe even separate methods `ASMmxBase::establishFirstBasis()` are in place since that seems to be a frequent use of this.

Reply:
for this one i'd prefer to keep the temporary, due to the refine basis stuff coming in a later commit. but i have made it explicitly typed and i only grab the front one. i don't think it's worth doing a separate function as the only extra work happening is that we push surf the 'surf' pointer at the end of the vector.